R/XcmsExperiment-functions.R

```diff
@@ -793,7 +793,9 @@
                                                  "largest_bpi"),
                                    msLevel = 2L, expandRt = 0, expandMz = 0,
                                    ppm = 0, skipFilled = FALSE,
-                                   peaks = integer(), BPPARAM = bpparam()) {
+                                   peaks = integer(),
+                                   chromPeakColumns = c("rt", "mz"),
+                                   BPPARAM = bpparam()) {
     method <- match.arg(method)
     pks <- .chromPeaks(x)[, c("mz", "mzmin", "mzmax", "rt",
```

Comment:
you need to add a check whether there are all `chromPeakColumns %in% colnames(.chromPeaks(x))`. Also, then you should ensure that these columns are also returned in following call, i.e. `pks <- .chromPeaks(x)[, union(c("mz", "mzmin", "mzmax", "rt", "rtmin", "rtmax", "maxo", "sample"), chromPeakColumns)]`